src/twisted/protocols/test/test_tls.py

```diff
@@ -977,6 +981,54 @@ def disconnected(ign):
         return disconnectDeferred
 
 
+    def test_noCircularReferences(self):
+        """
+        TLSMemoryBIOProtocol doesn't leave circular references that make
+        it alive after connection is closed.
+        """
+        def nObjectsOfType(type):
+            return sum(1 for x in gc.get_objects() if isinstance(x, type))
+
+        gc.disable()
+
+        try:
+            class DummyServerProtocol(Protocol):
+                def dataReceived(self, data):
+                    self.transport.loseConnection()
+
+
+            class DummyClientProtocol(Protocol):
```

Comment:
Again, why dummy?
This looks like a fully functional protocol which will ignore all received data and will send a `hello` message on a connection.